test/partiql-tests-runner/src/main/kotlin/org/partiql/runner/ConformanceComparison.kt

```diff
@@ -11,91 +11,208 @@ fun main(args: Array<String>) {
+
+    val firstTotalSize = firstPassingSize + firstFailingSize + firstIgnoreSize
+    val secondTotalSize = secondPassingSize + secondFailingSize + secondIgnoreSize
+
+    val firstPassingPercent = firstPassingSize.toDouble() / firstTotalSize * 100
+    val secondPassingPercent = secondPassingSize.toDouble() / secondTotalSize * 100
+
+    abstract val reportTitle: String
+    abstract fun generateComparisonReport(): String
+}
 
-    comparisonReportFile.writeText(
-        """### Conformance comparison report
-| | Base ($origCommitId) | $newCommitId | +/- |
+class CrossCommitReportAnalyzer(private val first: Report, private val second: Report) :
+    ReportAnalyzer(first, second) {
+    override val reportTitle: String = "Conformance comparison report-Cross Commit-${first.engine.uppercase()}"
+    override fun generateComparisonReport() =
+        buildString {
+            this.appendLine(
+                """### $reportTitle
+| | Base (${first.commitId}) | ${second.commitId} | +/- |
 | --- | ---: | ---: | ---: |
-| % Passing | ${"%.2f".format(origPassingPercent)}% | ${"%.2f".format(newPassingPercent)}% | ${"%.2f".format(newPassingPercent - origPassingPercent)}% |
-| :white_check_mark: Passing | $numOrigPassing | $numNewPassing | ${numNewPassing - numOrigPassing} |
-| :x: Failing | $numOrigFailing | $numNewFailing | ${numNewFailing - numOrigFailing} |
-| :large_orange_diamond: Ignored | $numOrigIgnored | $numNewIgnored | ${numNewIgnored - numOrigIgnored} |
-| Total Tests | $totalOrig | $totalNew | ${totalNew - totalOrig} |
-""",
-    )
-
-    comparisonReportFile.appendText(
-"""
+| % Passing | ${"%.2f".format(firstPassingPercent)}% | ${"%.2f".format(secondPassingPercent)}% | ${"%.2f".format(secondPassingPercent - firstPassingPercent)}% |
+| :white_check_mark: Passing | $firstPassingSize | $secondPassingSize | ${secondPassingSize - firstPassingSize} |
+| :x: Failing | $firstFailingSize | $secondFailingSize | ${secondFailingSize - firstFailingSize} |
+| :large_orange_diamond: Ignored | $firstIgnoreSize | $secondIgnoreSize | ${secondIgnoreSize - firstIgnoreSize} |
+| Total Tests | $firstTotalSize | $secondTotalSize | ${secondTotalSize - firstTotalSize} |
+                """.trimIndent()
+            )
+            this.appendLine(
+                """
 Number passing in both: ${passingInBoth.count()}
 
 Number failing in both: ${failingInBoth.count()}
 
-Number passing in Base ($origCommitId) but now fail: ${passingOrigFailingNew.count()}
+Number passing in Base (${first.commitId}) but now fail: ${passingFirstFailingSecond.count()}
+
+Number failing in Base (${first.commitId}) but now pass: ${failureFirstPassingSecond.count()}
+                """.trimIndent()
+            )
+            if (passingFirstFailingSecond.isNotEmpty()) {
+                // character count limitation with comments in GitHub
+                // also, not ideal to list out hundreds of test names
+                if (passingFirstFailingSecond.size < 10) {
```

Comment:
Perhaps we could still list out the test regressions in the conformance report even if there are many (at least for croos-commit)? GitHub PR comments still have a character limit but I don't believe the CI step to print the report in the run ([source](https://github.com/partiql/partiql-lang-kotlin/blob/92642b69a856860b166ee7e98da4abac7578b48d/.github/workflows/conformance-report.yml#L104-L106)) has a character limit. Just thinking it could still be helpful to print out all the failing tests.

Reply:
Similarly for the previously failing but now passing tests.

Reply:
Looked into the `$GITHUB_STEP_SUMMARY` a bit more. There is a limit of 1MiB per the [docs](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#step-isolation-and-limits). From the [other comment](https://github.com/partiql/partiql-lang-kotlin/pull/1333#pullrequestreview-1816250126), the size limit shouldn't be an issue as we improve the conformance of the eval engine. In the future if we were to compare more engines, we could create separate comparison reports between engines that we publish as separate steps.

All that said, we could also just download the comparison report from the published CI artifact if we need the full test diff.